### PR TITLE
Refactor typescript code

### LINKED ISSUE #1755: Typescript fixes for ComponentClass not including children
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `react-native-svg@12.3.0` for the project I'm working on.

I don't totally understand the reasons, but React.ComponentClass doesn't seem to include children anymore, so switching to ReactNode seems to make things right. ComponentClass<PropsWithChildren<X>> would work too, but not sure which is more accurate.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/react-native-svg/src/index.d.ts b/node_modules/react-native-svg/src/index.d.ts
index 891a99d..8f5df5e 100644
--- a/node_modules/react-native-svg/src/index.d.ts
+++ b/node_modules/react-native-svg/src/index.d.ts
@@ -265,8 +265,8 @@ export type Ellipse = React.ComponentClass<EllipseProps>;
 export interface GProps extends CommonPathProps, FontProps {
   opacity?: NumberProp;
 }
-export const G: React.ComponentClass<GProps>;
-export type G = React.ComponentClass<GProps>;
+export const G: React.ReactNode<GProps>;
+export type G = React.ReactNode<GProps>;
 
 export interface ForeignObjectProps {
   x?: NumberProp;
@@ -420,8 +420,8 @@ export interface TSpanProps extends CommonPathProps, FontProps {
   rotate?: NumberArray;
   inlineSize?: NumberProp;
 }
-export const TSpan: React.ComponentClass<TSpanProps>;
-export type TSpan = React.ComponentClass<TSpanProps>;
+export const TSpan: React.ReactNode<TSpanProps>;
+export type TSpan = React.ReactNode<TSpanProps>;
 
 export interface TextSpecificProps extends CommonPathProps, FontProps {
   alignmentBaseline?: AlignmentBaseline;
@@ -442,8 +442,8 @@ export interface TextProps extends TextSpecificProps {
   opacity?: NumberProp;
   inlineSize?: NumberProp;
 }
-export const Text: React.ComponentClass<TextProps>;
-export type Text = React.ComponentClass<TextProps>;
+export const Text: React.ReactNode<TextProps>;
+export type Text = React.ReactNode<TextProps>;
 
 export interface TextPathProps extends TextSpecificProps {
   xlinkHref?: string;
@@ -453,8 +453,8 @@ export interface TextPathProps extends TextSpecificProps {
   spacing?: TextPathSpacing;
   midLine?: TextPathMidLine;
 }
-export const TextPath: React.ComponentClass<TextPathProps>;
-export type TextPath = React.ComponentClass<TextPathProps>;
+export const TextPath: React.ReactNode<TextPathProps>;
+export type TextPath = React.ReactNode<TextPathProps>;
 
 export interface UseProps extends CommonPathProps {
   xlinkHref?: string;
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

### PR DESCRIPTION
**Pull request by codebutler**
# Summary

* Merged duplicate type definitions into implementation code
* Added explicit `children` props (fixes #1755, #1745)
* Added CI job to validate types

## Test Plan

New CI job should pass.

## Compatibility

No functional changes.

## Checklist

- [ ] I have tested this on a device and a simulator
- [ ] I added documentation in `README.md`
- [X] I updated the typed files (typescript)
- [ ] I added a test for the API in the `__tests__` folder

### CODE REVIEW

**Comment by codebutler** [comment]
@WoLewicki looks like quite an issue/PR backlog has piled up. would you like help maintaining this library?

**Comment by mhyassin** [comment]
I think this should be indeed merged in

**Comment by jozsef-ftrc** [comment]
This is the only typescript issue we've got with React Native 0.69 upgrade. +1

File: src/elements/Stop.tsx [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import { Component } from 'react';
+import { Color, NumberProp } from '../lib/extract/types';
 
-type StopProps = {
+export interface StopProps {
+  stopColor?: Color;
```
Comment (WoLewicki) [inline src/elements/Stop.tsx:5]: Where do these new props come from?

File: src/elements/NativeComponents.tsx [hunk 2]
```diff
@@ -1,4 +1,12 @@
 import { requireNativeComponent as rnc } from 'react-native';
+import React from 'react';
+
+declare module 'react-native' {
```
Comment (WoLewicki) [inline src/elements/NativeComponents.tsx:4]: Why is it here?

**Review by WoLewicki — COMMENTED** [review]
Thanks for submitting this PR :tada: I added some comments. Please answer them.

**Comment by WoLewicki** [comment]
Closing since #1806 has been merged. Thanks for your contribution @codebutler !

**Pull request closed without merging**